vkcs/monitoring/resource_cloud_monitoring.go

```diff
@@ -0,0 +1,233 @@
+	resp.TypeName = req.ProviderTypeName + "_cloud_monitoring"
+}
+
+type сloudMonitoringResource struct {
+	config clients.Config
+}
+
+type CloudMonitoringResourceModel struct {
+	ID     types.String `tfsdk:"id"`
+	Region types.String `tfsdk:"region"`
+
+	ImageID       types.String `tfsdk:"image_id"`
+	Script        types.String `tfsdk:"script"`
+	ServiceUserID types.String `tfsdk:"service_user_id"`
+}
+
+func (r *сloudMonitoringResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "ID of the resource.",
+			},
+
+			"region": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplaceIf(planmodifiers.GetRegionPlanModifier(resp),
+						"require replacement if configuration value changes", "require replacement if configuration value changes"),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Description: "The region in which to obtain the service client. If omitted, the `region` argument of the provider is used.",
+			},
+
+			"image_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplaceIfConfigured(),
+				},
+				Description: "ID of the image to create cloud monitoring for.",
+			},
+
+			"script": schema.StringAttribute{
+				Computed:    true,
+				Sensitive:   true,
+				Description: "Shell script of the cloud monitoring.",
+			},
+
+			"service_user_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "ID of the service monitoring user.",
+			},
+		},
+		Description: "Receives settings for cloud monitoring for the `vkcs_compute_instance'.",
+	}
+}
+
+func (r *сloudMonitoringResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.config = req.ProviderData.(clients.Config)
+}
+
+func (r *сloudMonitoringResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data CloudMonitoringResourceModel
+	diags := req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	region := data.Region.ValueString()
+	if region == "" {
+		region = r.config.GetRegion()
+	}
+
+	projectID := r.config.GetTenantID()
+
+	capabilitiesClient, err := r.config.CapabilityV1Client(region)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VKCS capabilities client", err.Error())
+		return
+	}
+
+	monitoringClient, err := r.config.CloudMonitoringV2Client(region)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VKCS cloud monitoring client", err.Error())
+		return
+	}
+
+	err = icapabilities.CheckAvailability(capabilitiesClient).Err
```

Comment:
You should place this in client creation function and add checks for both clients.